### ISSUE TITLE
Accessibility errors cross default backstage pages

### ISSUE THREAD

**Issue by linzhaoken** [backstage/backstage#7124]
My team start using backstage to build an internal application, because the application will be widely used by all developers in my company, I would like it to be accesisble and follow AA standard. 

Current I did a accessibility scan with axe tool on backstage default pages, I foud some error appears those errors are critical and serious. Below are a list of errors that I found. I can also provide csv files for full error report if needed.

**Catalog page**
1 critical: missing aria-label
8 serious: button inside button

**Filter component**
7 critical: 
Missing aria-label
Missing aria parent, menu, menu item
Form label are missing

**API doc page**
11 critical, icon button missing aria-label
14 serious,
  -  color contrast
- Missing aria-label
- Button in button

**Documentation** 
1 critical, missing aria-label
2 serious, color contrast, missing aria-label

**Create a new component page**
Filter error, similar as above

**New Service Form**
Missing aria-label, 2

**Comment by benjdlambert** [comment]
Hey :wave: we'd happily accept some of these as contributions, is this something you'd like to take on?

**Comment by linzhaoken** [comment]
@benjdlambert yes, I would like to contribute this as I am doing accessibility scan.

**Comment by benjdlambert** [comment]
Assigned! @linzhaoken!

**Comment by mhaley37** [comment]
Hi, 
We ran some accessibility scanning and testing and found some additional areas for improvement, including:

* The `SUPPORT` button in upper right hand corner of the Software Catalog -- when clicked -- launches a popover with some `moderate` accessibility concerns:
<img width="1746" alt="image" src="https://user-images.githubusercontent.com/2686352/161285620-b888918f-df8a-4ca9-99d6-d691c1a45fb2.png">

* On smaller screens, the filter `FILTERS` in the software catalog launch a left side drawer has similar accessibility issues concerning lack of landmarks (or content outside of landmarks)

* The `OAuthRequestDialog` component from @backstage\core-components ( it is created as a [Dialogue](https://mui.com/components/dialogs/) ) does not use`aria-modal:"true"`, and instead uses `aria-hidden:true` to hide layers underneath when the dialogue has focus.  This triggers some landmark warnings.

Could changes to address these be included in this work?  Thanks.

**Comment by benjdlambert** [comment]
@mhaley37 I think this work has gone stale, so maybe you would like to contribute some of these issues? :pray:

**Comment by mhaley37** [comment]
HI @benjdlambert, sure!  I discussed this with our team and we'd be happy to contribute to making a more accessible platform.  You can assign to me and I should have some bandwidth soon and can contribute some work.  Thanks

**Comment by benjdlambert** [comment]
Done!

**Comment by mhaley37** [comment]
https://github.com/backstage/backstage/pull/11591 fixes a number of accessibility issues in the catalog and component pages by disabling draggability on table columns.
